ast/src/parsed/display.rs

```diff
@@ -681,6 +685,55 @@ impl<E: Display> Display for EnumVariant<E> {
     }
 }
 
+impl<E: Display> Display for StructDeclaration<E> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        let name = self.name.to_string();
+        let type_vars = if self.type_vars.is_empty() {
+            Default::default()
+        } else {
+            format!("<{}>", self.type_vars)
+        };
+        write!(
+            f,
+            "struct {name}{type_vars} {{\n{}}}",
+            indent(
+                self.fields
+                    .iter()
+                    .map(|v| format!("{}: {},\n", v.0, v.1))
```

Comment:
```suggestion
                    .map(|(name, ty)| format!("{name}: {ty},\n"))
```

Reply:
Didn't we want to re-use some struct from the trait definition here?

Reply:
`TraitFunction` would be the ideal option, so I think we should rename them: https://github.com/powdr-labs/powdr/pull/1816